crypto/keybase/keybase.go

```diff
@@ -1,40 +1,29 @@
 package keybase
 
 import (
-	"fmt"
-
-	clkeys "github.com/cosmos/cosmos-sdk/client/keys"
-	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	hd "github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// GenerateSaveCoinKey returns the address of a public key, along with the secret
+// SaveCoinKey returns the address of a public key, along with the secret
 // phrase to recover the private key.
-
 func SaveCoinKey(clientRoot, keyName, keyPass, keyMnemonic string, overwrite bool) (sdk.AccAddress, string, error) {
-	// get the keystore from the client
-	keybase, err := clkeys.NewKeyBaseFromDir(clientRoot)
-	if err != nil {
-		return []byte{}, "", err
-	}
-
-	// ensure no overwrite
-	if !overwrite {
-		_, err := keybase.Get(keyName)
-		if err == nil {
-			return []byte{}, "", fmt.Errorf(
-				"key already exists, overwrite is disabled (clientRoot: %s)", clientRoot)
-		}
-	}
+	// TODO
+	// kn-sdk? appName
+	kr, err := keyring.New("kn-sdk", keyring.BackendPass, clientRoot, nil)
```

Comment:
kr may have nil if err is not nil

Reply:
There must be error checking code on the 15 line, before calling kr.NewMnemonic or kr.NewAccount functions